src/world_interface/kinematic_common_interface.cpp

```diff
@@ -170,9 +170,23 @@ double getJointPowerValue(types::jointid_t joint) {
 
 void setJointMotorPower(robot::types::jointid_t joint, double power) {
 	using robot::types::jointid_t;
-
 	if (jointMotorMap.find(joint) != jointMotorMap.end()) {
-		setMotorPower(jointMotorMap.at(joint), power);
+		// TODO: check if associated motor is in IK_MOTORS, don't check specifically these ones
+		if (joint == jointid_t::elbow || joint == jointid_t::shoulder) {
+			if (!Globals::armIKEnabled) {
+				setMotorPower(jointMotorMap.at(joint), power);
+			}
+		} else {
+			setMotorPower(jointMotorMap.at(joint), power);
+		}
```

Comment:
This set of nested if-else can just be simplified to "if ik not enabled or this isn't an ik joint", i.e.
```c++
if (!Globals::armIKEnabled || !std::find(IK_MOTORS.begin(), IK_MOTORS.end(), joint)) {
    setMotorPower(...);
}
```
which is cleaner